app/javascript/components/search/controls/download/DownloadSelectionModal.js

```diff
@@ -43,41 +43,51 @@ export default function DownloadSelectionModal({ studyAccessions, tdrFileInfo, s
     setDownloadInfoTDR(tdrFileInfo)
     setIsLoadingTDR(false)
     // pull the drs ids out of each study and put them in a single array
+    // calling filter at end removes metadata file that does not have a DRS id
     const drsIds = tdrFileInfo.map(study => study.studyFiles.map(sfile => sfile.drs_id))
-      .reduce((acc, val) => acc.concat(val), [])
+      .reduce((acc, val) => acc.concat(val), []).filter(e => e != null)
```

Comment:
Strict equality is conventional:

```suggestion
      .reduce((acc, val) => acc.concat(val), []).filter(e => e !== null)
```